src/main/java/org/springframework/integration/kafka/inbound/KafkaMessageDrivenChannelAdapter.java

```diff
@@ -55,9 +59,16 @@ public void setPayloadDecoder(Decoder<?> payloadDecoder) {
 		this.payloadDecoder = payloadDecoder;
 	}
 
+	public void setAutoCommitOffset(boolean autoCommitOffset) {
+		this.autoCommitOffset = autoCommitOffset;
+	}
+
 	@Override
 	protected void onInit() {
-		this.messageListenerContainer.setMessageListener(new ChannelForwardingMessageListener());
+		this.messageListenerContainer.setAutoCommitOffset(autoCommitOffset);
```

Comment:
That's a good idea. I had the flag initially and the check later, but they can be collapsed into one. That'd be much cleaner for sure.